qiskit/ml/algorithms/quantum_kernel.py

```diff
@@ -0,0 +1,272 @@
+        """
+        Args
+            feature_map:       # parametrised circuit to be used as the feature map
+            enforce_psd:       # project to closest positive semidefinite matrix if x = y
+            quantum_instance:  # Quantum Instance or Backend
+        """
+        # super().__init__(quantum_instance)
+
+        self._feature_map = feature_map
+        self._enforce_psd = enforce_psd
+        self._quantum_instance = quantum_instance
+
+    @property
+    def feature_map(self) -> QuantumCircuit:
+        """ Returns feature map """
+        return self._feature_map
+
+    @feature_map.setter
+    def feature_map(self, feature_map: QuantumCircuit):
+        """ Sets feature map """
+        self._feature_map = feature_map
+
+    @property
+    def quantum_instance(self) -> QuantumInstance:
+        """ Returns quantum instance """
+        return self._quantum_instance
+
+    @quantum_instance.setter
+    def quantum_instance(self, quantum_instance: Union[Backend,
+                                                       BaseBackend, QuantumInstance]) -> None:
+        """ Sets quantum instance """
+        if isinstance(quantum_instance, (BaseBackend, Backend)):
+            self._quantum_instance = QuantumInstance(quantum_instance)
+        else:
+            self._quantum_instance = quantum_instance
+
+    def _construct_circuit(self, x, y, is_statevector_sim=False):
+        """
+        Helper function to generate inner product circuit for the given feature map.
+        """
+        q = QuantumRegister(self._feature_map.num_qubits, 'q')
+        c = ClassicalRegister(self._feature_map.num_qubits, 'c')
+        qc = QuantumCircuit(q, c)
+
+        x_dict = dict(zip(self._feature_map.parameters, x))
+        psi_x = self._feature_map.assign_parameters(x_dict)
+        qc.append(psi_x.to_instruction(), qc.qubits)
+
+        if not is_statevector_sim:
+            y_dict = dict(zip(self._feature_map.parameters, y))
+            psi_y_dag = self._feature_map.assign_parameters(y_dict)
+            qc.append(psi_y_dag.to_instruction().inverse(), qc.qubits)
+
+            qc.barrier(q)
+            qc.measure(q, c)
+        return qc
+
+    def _compute_overlap(self, idx, results, is_statevector_sim):
+        """
+        Helper function to compute overlap for given input.
+        """
+        if is_statevector_sim:
+            i, j = idx
+            v_a = results.get_statevector(int(i))
+            v_b = results.get_statevector(int(j))
+            # |<0|Psi^daggar(y) x Psi(x)|0>|^2, take the amplitude
+            tmp = np.vdot(v_a, v_b)
+            kernel_value = np.vdot(tmp, tmp).real  # pylint: disable=no-member
+        else:
+            result = results.get_counts(idx)
+
+            measurement_basis = '0' * self._feature_map.num_qubits
+            kernel_value = result.get(measurement_basis, 0) / sum(result.values())
+        return kernel_value
+
+    def evaluate(self, x_vec, y_vec=None):
```

Comment:
Can we move typehints to the function definition out of the docstrings

Reply:
Done in 2edf212